### PR TITLE
fix: address bugs with tasks logic found after deployment

### PR DESCRIPTION
**Pull request by bcoe**
A few things that needed to be corrected:

1. logic for enqueuing fails unless `content-type` set to `application/json`.
2. status needs to be set in response (or else it's always a `200`).
3. `request.body` was already a string, so we were double serializing.

### CODE REVIEW

File: packages/gcf-utils/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gcf-utils",
-  "version": "3.0.1",
+  "version": "3.0.3",
```
Comment (JustinBeckwith) [inline packages/gcf-utils/package.json:3]: That's interesting.  Are we manually bumping versions in this package?

File: packages/gcf-utils/src/gcf-utils.ts [hunk 2]
```diff
@@ -104,17 +104,16 @@ export class GCFBootstrapper {
         request.get('x-github-delivery') ||
         request.get('X-GitHub-Delivery') ||
         '';
-      const delivery =
-        request.get('x-hub-signature') || request.get('X-Hub-Signature') || '';
+      // TODO: add test to validate this signature is used on the initial
```
Comment (JustinBeckwith) [inline packages/gcf-utils/src/gcf-utils.ts:107]: Can we up-level this to a GitHub issue so we don't lose it?

File: packages/gcf-utils/package.json (hunk 1 above)
Comment (bcoe) [reply packages/gcf-utils/package.json:3]: yes, release-please doesn't know how to deal with this package structure. I think the best solution would be to extract `gcf-utils` form the mono-repo.

File: packages/gcf-utils/src/gcf-utils.ts (hunk 2 above)
Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts:107]: done: https://github.com/googleapis/repo-automation-bots/issues/470

File: packages/gcf-utils/src/gcf-utils.ts [hunk 3]
```diff
@@ -104,17 +104,14 @@ export class GCFBootstrapper {
...
+      // webhook from GitHub:
       const signature =
-        request.get('x-github-delivery') ||
-        request.get('X-GitHub-Delivery') ||
-        '';
+        request.get('x-hub-signature') || request.get('X-Hub-Signature') || '';
```
Comment (orthros) [inline packages/gcf-utils/src/gcf-utils.ts:110]: I'm not sure this is correct. Consider these lines
https://github.com/probot/serverless-gcf/blob/master/index.js#L39-L50. I believe the "signature" does need to be "x-github-delivery" and we just named things poorly.

Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts:110]: works for me, I will address.

Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts]: @orthros okay, reverted the header, @sofisl is testing everything out.

**Pull request merged**